### PR TITLE
Feature/index var in map

### PR DESCRIPTION
**Pull request by martinv13**
Bonjour @Gmousse ,
I have been working on some features aimed at having `.dropDuplicates()` working on a subset of columns. Here's what I've done:

- Add row index variable to `iter` function, thus enabling to get row index in `.map()`, `.chain()` and `.filter()`;
- Create these functions on `GroupedDataFrame`, with the row index being the row index within each group (these functions return new `DataFrame`s); 
- Change `.dropDuplicates()` which now just calls `.filter((row, i) => i === 0)` on the appropriately grouped `DataFrame`

What are your thoughts on that?
Thanks, Martin

### CODE REVIEW

**Comment by Gmousse** [comment]
@martinv13 It looks promising ! 
I will checkout your pr to play with tests, but I didn't see nothing strange in the code.

File: src/dataframe.js [hunk 1]
```diff
@@ -775,9 +775,10 @@ class DataFrame {
      * @example
```
Comment (Gmousse) [inline src/dataframe.js]: Just add the description of the new parameter columnNames (to show it in the documentation).
Then, it's fine :D

**Comment by Gmousse** [comment]
@martinv13 Well done. Just add the comment on doctring for dropDuplicates and resolve merges (you can juste merge develop and consider using your version of these files).
When it's done, I will merge.

**Comment by martinv13** [comment]
@Gmousse I guess it should be ok now!

**Comment by Gmousse** [comment]
@martinv13 well done

**Pull request merged**